nav2_costmap_2d/test/unit/footprint_collision_checker_test.cpp

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2020 Shivang Patel
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <vector>
+#include <memory>
+
+#include "gtest/gtest.h"
+#include "nav2_costmap_2d/footprint_collision_checker.hpp"
+
+TEST(collision_footprint, footprintCostWithPose)
+{
+  std::shared_ptr<nav2_costmap_2d::Costmap2D> costmap_ =
+    std::make_shared<nav2_costmap_2d::Costmap2D>(100, 100, 0.1, 0, 0, 0);
+
+  geometry_msgs::msg::Point p1;
```

Comment:
This test is all zeros, so of course it will work. Add obstacles around the robot and then try that. My recommendation:
- Add a cost 1 cell to the right in the middle
- Add a cost cell to the left on a corner
- Test that in this pose its OK
- Test that when you move the footprint 1 cell to the left its in collision (checks point cost)
- Test that when you move the footprint 1 cell to the right its in collision (checks line cost)